### DISCUSSION TITLE
Mailer: fgets(): SSL: Connection reset by peer

### DISCUSSION THREAD

**Discussion by maki3000** [symfony/symfony#48693] [Q&A]
### Symfony version(s) affected

6.2

### Description

When using the SMTP option for sending email on my shared hosting, I get:
`fgets(): SSL: Connection reset by peer`

I know that this is very server related, but I hope, I find help here.

I'm not sure if this is a server issue, but sending emails over e.g. Mac Mail, which uses the SMTP settings works. What is also strange, is that on the development environment it works, only the live server has issues.

### How to reproduce

Here are my SMTP env settings:
```
MAIL_MAILER=smtp
MAIL_HOST=lx59.hoststar.hosting
MAIL_PORT=465
MAIL_USERNAME=noreply@goodbad.world
MAIL_PASSWORD="{password}"
MAIL_ENCRYPTION=tls
MAIL_FROM_ADDRESS="noreply@goodbad.world"
MAIL_FROM_NAME="GB-Test"
```
Here is the request:
```
curl "https://test.maki3000.net/laravel/public/mail" \
   -X GET \
   -H 'cookie: XSRF-TOKEN=eyJpdiI6IkZTSzNkZE8wVG5ZeG82VktsOFlxeEE9PSIsInZhbHVlIjoiNzREY0czcXFLeGt4Z2JZWitwU0NnQkNtOXRVaElQUUthTS9qQVBKZXVpaUFVN3RnQTIvNkRoTnFadnNJZVNYMmM3K1lLYmNoVU83QXo2YkhFK3FRbThPSHo4N05XMEdSMW9wTlRXWnpldHE4NjZiL1RYZUMwMXBxWjdjVVpUWFEiLCJtYWMiOiI1ODBlMzhjOWQzYjhlYzFjNGVmNmUwYTQ0OWY5YTBlNmQ3MDEyMzc2YmQwNjc5NTA2NWVmMmNhZmQ0YmE3Y2YxIiwidGFnIjoiIn0%3D; laravel_session=eyJpdiI6IkhsMWhzdHhhM1dIYUlYZ3VqN0RQL0E9PSIsInZhbHVlIjoibUNMSHdvcyt4TUZ2dFV3bWRjNE10UC9HVi9vREI1d0Q4QWo0Zmk0eUltclFmZVkyVGVhRHF0aThaem95U1BnNTJ2UEVQa0dnRWw0ZGhqUnJVSUVDK0gxL1lpamRnL0pJY3dNakNmay9kZUhrSzd3cUdzV0V2djRWdXVqRFgyQVMiLCJtYWMiOiI4OWNmYWVjYzk1ZjYzN2U5YzgyMTBkZjU4ZTE0NDA0YTdjMzY0M2IxMjRhNTFjNGJjZjZlMDE2YjEwODI3NmNkIiwidGFnIjoiIn0%3D' \
   -H 'accept-language: de-DE,de;q=0.9,en-US;q=0.8,en;q=0.7' \
   -H 'accept-encoding: gzip, deflate, br' \
   -H 'sec-fetch-dest: document' \
   -H 'sec-fetch-user: ?1' \
   -H 'sec-fetch-mode: navigate' \
   -H 'sec-fetch-site: none' \
   -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9' \
   -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36' \
   -H 'upgrade-insecure-requests: 1' \
   -H 'sec-ch-ua-platform: "macOS"' \
   -H 'sec-ch-ua-mobile: ?0' \
   -H 'sec-ch-ua: "Not?A_Brand";v="8", "Chromium";v="108", "Google Chrome";v="108"' \
   -H 'connection: close' \
   -H 'x-real-ip: 83.76.18.207' \
   -H 'host: test.maki3000.net';
```

### Possible Solution

My hosting support suggested that it could be the top level domains. Top level domains like hosting or world could be unsupported.

**Comment by maki3000** [comment]
I got it working by adding the `local_domain` with the IP of my hosters server address. Something like this:
`MAILER_DSN=smtp://mail@example.com:password@lx59.hoststar.hosting:465?local_domain=157.90.84.138`

**Reply by 12bPilot** [reply]
Running into the same problem, also using a hoststar shared hosting account.
Did you add that MAILER_DSN entry to the existing MAIL_MAILER, etc entries in the .env file? Or replaced all the MAIL_*** entries with that MAILER_DSN line?

**Reply by maki3000** [reply]
I'm using usually October CMS and I made a post about it here:
https://talk.octobercms.com/t/possible-solution-for-smtp-failing-fgets-ssl-connection-reset-by-peer/855

**Reply by 12bPilot** [reply]
Thank you for the quick reply.
Using the local domain as you suggested in that post did the trick! Great solution, thanks!
